flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -231,6 +231,11 @@ public Builder writeParallelism(int newWriteParallelism) {
       return this;
     }
 
+    public Builder properties(Map<String, String> properties) {
+      upsert(PropertyUtil.propertyAsBoolean(properties, UPSERT_ENABLED, UPSERT_ENABLED_DEFAULT));
+      return this;
+    }
```

Comment:
As a follow up, we should consider supporting more options from in-line SQL comments.

Is this `properties` needed for any SQL in-line comment based configuration or just for our own custom configurations that Flink doesn't know about?

Reply:
Yes, in addition to `upsert`, we have more configuration that can be used in inline SQL , such as write file size and so on.
For more configuration that can be supported, see FlinkWriteConf.java of  #3998.
I'd prefer to wait for #3998 to finish.